src/parquet/column/reader.cc

```diff
@@ -33,7 +33,10 @@ ColumnReader::ColumnReader(const ColumnDescriptor* descr,
   : descr_(descr),
     pager_(std::move(pager)),
     num_buffered_values_(0),
-    num_decoded_values_(0) {}
+    num_decoded_values_(0) {
+      repetition_level_decoder_.Init(descr->max_repetition_level());
+      definition_level_decoder_.Init(descr->max_definition_level());
```

Comment:
Essentially, the upper layers should have the flexibility to create an instance or reset the state of the instance.